stix_shifter_modules/elastic_ecs/stix_translation/query_constructor.py

```diff
@@ -304,6 +305,77 @@ def _format_translated_queries(query_array):
     return formatted_queries
 
 
+def _unfold_case_insensitive_regex(regex_pattern):
+    # this function should be executed after "\s" unfolding
+
+    if "(?i)" in regex_pattern:
+
+        escaped_left_bracket_symbol = "辶"
```

Comment:
Are these the characters you meant to include? They don't look alphanumeric.

Reply:
Yes, this is a hack to distinguish square bracket and escaped square bracket in a regex. I intentionally use the non-ascii unicode symbols, which should not exist in any dataset (partial Chinese char, even not a valid Chinese char).

Reply:
More background: chars inside square bracket of a regex should be handled differently than chars outside square brackets. However, recognizing square bracket is tricky since it may be escaped, and the escape symbol (backslash) may also be escaped so `\[` may not mean escaped square bracket if the char before these substring is `\` like `hello\\[hi`. The unicode with split/join is the hack to deal with these.